Assets/Scripts/InteractionDetect.cs

```diff
@@ -103,4 +103,30 @@ void Update()
             }
         }
     }
+
+    // Returns 'true' if we touched or hovering on Unity UI element.
+    public static bool IsPointerOverUIElement()
+    {
+        return IsPointerOverUIElement(GetEventSystemRaycastResults());
+    }
+    // Returns 'true' if we touched or hovering on Unity UI element.
+    public static bool IsPointerOverUIElement(List<RaycastResult> eventSystemRayCastResults)
+    {
+        for (int index = 0;  index < eventSystemRayCastResults.Count; index ++)
+        {
+            RaycastResult curRaysastResult = eventSystemRayCastResults [index];

```

Comment:
```suggestion
            RaycastResult curRaycastResult = eventSystemRayCastResults [index];

```
Also there's some inconsistency between `...Raycast...` and `...RayCast...` but it may not be worth doing anything about at this point.

Reply:
Thanks for catching these!  `Raycast` is correct.  I'll add changes for all of your notes.